doc/rtd/topics/bugs.rst

```diff
@@ -0,0 +1,107 @@
+.. _reporting_bugs:
+
+Reporting Bugs
+**************
+
+The following documents:
+
+1) How to collect information for reporting bugs
+2) How to file bugs to the upstream cloud-init project or for distro specific
+   packages
+
+Collect Logs
+============
+
+To aid in debugging, please collect the necessary logs. To do so, run the
+`collect-logs` subcommand to produce a tarfile that you can easily upload:
+
+.. code-block:: shell-session
+
+  $ cloud-init collect-logs
+  Wrote /home/ubuntu/cloud-init.tar.gz
+
+If your version of cloud-init does not have the  `collect-logs` subcommand,
+then please manually collect the base log files by doing the following:
+
+.. code-block:: shell-session
+
+  $ dmesg > dmesg.txt
+  $ sudo journalctl -o short-precise > journal.txt
+  $ sudo tar -cvf cloud-init.tar dmesg.txt journal.txt /run/cloud-init \
+      /var/log/cloud-init.log /var/log/cloud-init-output.log
+
+Report Upstream Bug
+===================
+
+Bugs for upstream cloud-init are tracked using Launchpad. To file a bug:
+
+1. Collect the necessary debug logs as described above
+2. `Create a Launchpad account`_ or login to your existing account
+3. `Report an upstream cloud-init bug`_
+
+If debug logs are not provided, you will be asked for them before any
+further time is spent debugging.
```

Comment:
Should we add a proviso that if people can't collect these then they should mention that? I don't think we want people to not file bugs because things have ended up so broken that they can't get the logs, but maybe we should expect people to work that out themselves.

Reply:
I am ok with adding a comment about not being able to get them that they should mention it. Let me do that real quick.

Reply:
Added